### DISCUSSION TITLE
Splitting CoveragePlot() with integrated multiome object

### DISCUSSION THREAD

**Discussion by jjwestfall521** [stuart-lab/signac#796] [Q&A] [answered]
Hi @timoast,

I have an integrated (2 samples) multiomic (scRNA+scATAC) Seurat object, generated with data from the 10X Genomics multiome kit.

I integrated this object with anchors from the SCT assay and was able to use MACS2 to call peaks. I then ran `FindMultiModalNeighbors()` and `FindClusters()` to cluster my data.

Before integration, I added metadata such that the integrated object `combo.1hr@meta.data[["sample"]]` contains my sample names, "PTZ_1hr" and "Saline_1hr"

My goal is to generate a coverage plot shows me the chromatin accessibility of one or more clusters, split between my samples.

When I run:
```
CoveragePlot(
  object = combo.1hr,
  region = "Pdgfra",
  features = "Pdgfra",
  expression.assay = "SCT",
  extend.upstream = 10000,
  extend.downstream = 10000,
  idents = "10"
)
```
I get:

![image](https://user-images.githubusercontent.com/76490974/133482438-bd31ea1d-6398-40a0-99bf-7c80f884736a.png)
 
Where cluster "10" are my cells of interest. When I run:
```
CoveragePlot(
  object = combo.1hr,
  region = "Pdgfra",
  features = "Pdgfra",
  expression.assay = "SCT",
  extend.upstream = 10000,
  extend.downstream = 10000,
  group.by='sample'
)
```
I get:

![image](https://user-images.githubusercontent.com/76490974/133482841-055e6919-b639-497e-adf8-1a9032cb02bc.png)

which looks great, but shows the accessibility for all cells in each sample.

If I try and combine these:
```
CoveragePlot(
  object = combo.1hr,
  region = "Pdgfra",
  features = "Pdgfra",
  expression.assay = "SCT",
  extend.upstream = 10000,
  extend.downstream = 10000,
  idents = "10",
  group.by='sample'
)
```
I get the following error:
```
Error in WhichCells.Seurat(object = object, idents = idents) : 
  Cannot find the following identities in the object: 10
```

Is there a way to show the chromatin accessibility in both samples, but only for those cells in cluster 10? I know there is no `split.by` argument for `CoveragePlot()`. Is there a way to use the `cells` argument to generate separate graphs using the objects metadata? Then I could just use patchwork to wrap the plots together. I am unclear how to proceed.

Thanks for your help.

**Comment by timoast** [comment, marked as answer]
Hi @jjwestfall521, you can combine the sample and cluster information into a new metadata variable and then group the cells based on that. For example:

```r
combo.1hr$sample_cluster <- paste0(combo.1hr$sample, "_", combo.1hr$seurat_clusters)

CoveragePlot(
  object = combo.1hr,
  region = "Pdgfra",
  features = "Pdgfra",
  expression.assay = "SCT",
  extend.upstream = 10000,
  extend.downstream = 10000,
  idents = c("Saline_1hr_10", "PTZ_1hr_10"),
  group.by='sample_cluster'
)
```

**Reply by jjwestfall521** [reply]
That worked great! Thanks so much! 

![image](https://user-images.githubusercontent.com/76490974/133497138-e15f0f83-7b11-42f2-b441-98fe691d2a74.png)
